src/cli/cli.cpp

```diff
@@ -3835,12 +3835,26 @@ void Interpreter::ProcessUnsecurePort(uint8_t aArgsLength, char *aArgs[])
 
 void Interpreter::ProcessVersion(uint8_t aArgsLength, char *aArgs[])
 {
-    OT_UNUSED_VARIABLE(aArgsLength);
-    OT_UNUSED_VARIABLE(aArgs);
+    otError error = OT_ERROR_NONE;
 
-    const char *version = otGetVersionString();
-    OutputLine("%s", static_cast<const char *>(version));
-    OutputResult(OT_ERROR_NONE);
+    if (aArgsLength == 0)
+    {
+        const char *version = otGetVersionString();
+        OutputLine("%s", static_cast<const char *>(version));
```

Comment:
I know it's not part the change in this PR/commit, but since these lines are being included (and tabbed) in a new`if`, we can improve the code as well (I think there is no need to `static_cast` from same type and can inline the call).
```suggestion
        OutputLine("%s", otGetVersionString());
```